clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceChangelogConsumerImpl.java

```diff
@@ -155,28 +150,140 @@ public VeniceChangelogConsumerImpl(
+      List<TopicPartition> topicPartitionList =
+          getPartitionListToSubscribe(partitions, topicPartitionSet, topicToSubscribe);
       kafkaConsumer.assign(topicPartitionList);
+      for (TopicPartition topicPartition: topicPartitionList) {
+        if (!topicPartition.topic().endsWith(ChangeCaptureView.CHANGE_CAPTURE_TOPIC_SUFFIX)) {
+          compressorMap.put(topicPartition.partition(), getVersionCompressor(topicPartition));
+        }
+      }
       kafkaConsumer.seekToBeginning(topicPartitionList);
-      subscribedPartitions.addAll(partitions);
+      return null;
+    });
+  }
+
+  protected VeniceCompressor getVersionCompressor(TopicPartition topicPartition) {
+    Store store = storeRepository.getStore(storeName);
+    Version version = store.getVersion(Version.parseVersionFromVersionTopicName(topicPartition.topic())).get();
+    VeniceCompressor compressor;
+    if (CompressionStrategy.ZSTD_WITH_DICT.equals(version.getCompressionStrategy())) {
+      compressor = compressorFactory.getVersionSpecificCompressor(topicPartition.topic());
+      if (compressor == null) {
+        // we need to retrieve the dictionary from the kafka topic
+        ByteBuffer dictionary = DictionaryUtils.readDictionaryFromKafka(
+            topicPartition.topic(),
+            new VeniceProperties(changelogClientConfig.getConsumerProperties()));
+        compressor = compressorFactory.createVersionSpecificCompressorIfNotExist(
+            version.getCompressionStrategy(),
+            topicPartition.topic(),
+            dictionary.array());
+      }
+    } else {
+      compressor = compressorFactory.getCompressor(version.getCompressionStrategy());
+    }
+    return compressor;
+  }
+
+  @Override
+  public CompletableFuture<Void> seekToBeginningOfPush(Set<Integer> partitions) {
+    // Get latest version topic
+    storeRepository.refresh();
+    Store store = storeRepository.getStore(storeName);
+    int currentVersion = store.getCurrentVersion();
+    String topic = Version.composeKafkaTopic(storeName, currentVersion);
+    return internalSeek(partitions, topic, kafkaConsumer::seekToBeginning);
+  }
+
+  @Override
+  public CompletableFuture<Void> seekToBeginningOfPush() {
+    return seekToBeginningOfPush(
+        kafkaConsumer.assignment()
+            .stream()
+            .map(topicPartition -> topicPartition.partition())
+            .collect(Collectors.toSet()));
+  }
+
+  @Override
+  public CompletableFuture<Void> seekToEndOfPush(Set<Integer> partitions) {
+    // Get latest change capture topic
+    storeRepository.refresh();
+    Store store = storeRepository.getStore(storeName);
+    int currentVersion = store.getCurrentVersion();
+    String topic = Version.composeKafkaTopic(storeName, currentVersion) + ChangeCaptureView.CHANGE_CAPTURE_TOPIC_SUFFIX;
+    return internalSeek(partitions, topic, kafkaConsumer::seekToBeginning);
```

Comment:
Is it possible we will change `kafkaConsumer::seekToBeginning` here in future? Besides, I feel like the logic here is a bit complicated: `seekToBeginningOfPush` is only for version topic, `seekToEndOfPush` is only for change capture topic. If so, we might need to add explaination to the java docs of the API.

Reply:
In the interface javadoc or in the implementation?

Reply:
If this behavior I mentioned here is right, I think we need to add to the javadoc of API.

Reply:
I'm not so sure we should expose details of 'version topic' and 'real time' topic specifically to a customer facing interface, but I'll add some ntoes about this seeking to the start/end of batch pushes.